### PR TITLE
Customer personas (first draft)

### PR DESCRIPTION
**Pull request by teozkr**
Based on what we came to in today's meeting (https://docs.google.com/document/d/1K6BIvARKWpum1XJ2rveYbJGK4s1cyOMxnuCj_QyjKyU/edit#) but cleaned up slightly.

Created a new internal section for this stuff. Probably want to move the ADRs into it as well at some point, but I'm not really sure about whether this is the correct way to manage it.

### CODE REVIEW

File: modules/internal/pages/personas.adoc [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+Robot authentication::
+  Does the customer have an opinion on how servers authenticate each other? +
+  (For example: Kafka servers need to connect to ZooKeeper to maintain quorum, even if there are no Kafka clients at the time.)
+Network control::
+  DNS;;
+    Is the customer willing to delegate us a DNS zone, or are we "stuck" adapting to whatever they provision for us?
+  TLS;;
+    Does the customer want us to integrate into their existing TLS PKI? If so, is it automated? What are their issuing policies?
+  Kerberos;;
+    Does the customer want us to integrate into their existing Kerberos realm?
```
Comment (soenkeliebau) [inline modules/internal/pages/personas.adoc:19]: Maybe briefly mention trusts and if they are an option?

File: modules/internal/pages/personas.adoc [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+  Overlay peering/BGP;;
+    Is the Kubernetes overlay network peered into their regular network environment?
+  NIC bonding & multihoming;;
+    Do they use bonded NICs that we need to take into account?
+  Network-level firewalls;;
+    Do they use network-level firewalls that our traffic needs to be allowed through?
+  Linux Security Modules?;;
+    Do they require all machines to use SELinux, AppArmor, or something similar?
+Kubernetess cluster::
```
Comment (soenkeliebau) [inline modules/internal/pages/personas.adoc:28]:
```suggestion
Kubernetes cluster::
```

File: modules/internal/pages/personas.adoc [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+  Dedicated cluster;;
+    Can we expect to get a dedicated Stackable Kubernetes cluster, or will be be colocated with internal stuff?
+  Premises;;
+    Is it hosted in the cloud or on-premise?
+  Managed;;
+    Is it managed by an external service provider (IONOS, GKE, EKS, AKS, etc), by us, or is it managed by the customer themselves?
+  Node lifetime;;
+    How long can a single node identity be expected to last for? (Years, weeks, days, hours)
```
Comment (soenkeliebau) [inline modules/internal/pages/personas.adoc:36]: This mainly refers to the hostname of the node?

File: modules/internal/pages/personas.adoc [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+  Kerberos usage;;
+    Issued manually by customer
+  Overlay peering/BGP;;
+    None
+  NIC bonding & multihoming;;
+    Yes
+  Network-level firewalls;;
+    Port-based, manually configured 
+  Linux Security Modules? (SELinux/AppArmor);;
+    ???
```
Comment (soenkeliebau) [inline modules/internal/pages/personas.adoc:66]: For all intents and purposes we can assume "no" here I believe. I've not yet seen anybody follow through on SeLinux..
@lfrancke  ?

**Review by soenkeliebau — CHANGES_REQUESTED** [review]
Left some questions and one (nitpicking) correction. 

One general question, there is no distinction between inside/outside access to our services, we are assuming outside access is always necessary?

File: modules/internal/pages/personas.adoc (hunk 2 above)
Comment (teozkr) [reply modules/internal/pages/personas.adoc:28]: Oops.

File: modules/internal/pages/personas.adoc (hunk 3 above)
Comment (teozkr) [reply modules/internal/pages/personas.adoc:36]: Hostname, IP address, local disks, etc.

File: modules/internal/pages/personas.adoc (hunk 1 above)
Comment (teozkr) [reply modules/internal/pages/personas.adoc:19]: I don't think I'm familiar enough to write a reasonable question here, but feel free to suggest something.

File: modules/internal/pages/personas.adoc (hunk 4 above)
Comment (lfrancke) [reply modules/internal/pages/personas.adoc:66]: Agreed

Comment (teozkr) [reply modules/internal/pages/personas.adoc]: Ok, changed this to no then.

File: modules/internal/pages/personas.adoc (hunk 3 above)
Comment (teozkr) [reply modules/internal/pages/personas.adoc:36]: Added a note about this to the doc

File: modules/internal/pages/personas.adoc (hunk 1 above)
Comment (soenkeliebau) [inline modules/internal/pages/personas.adoc:19]:
```suggestion
    Does the customer want us to integrate into their existing Kerberos realm, if so is it an option to run our own kdc and integrate via trusts? 
```

**Review by soenkeliebau — APPROVED** [review]

**Comment by soenkeliebau** [comment]
paging Mr. Antora (aka @lfrancke ) regarding the conflict in the antora part of the pr :)

**Comment by lfrancke** [comment]
That's prerry straightforward. The antora.yml has changed. This just needs a manual resolve but that is easy.
We used to have multiple entries in there, now we only have one. Just add this new one at the end.

**Review by lfrancke — APPROVED** [review]

**Pull request merged**